src/RegistrationMsg.chpl

```diff
@@ -495,9 +462,20 @@ module RegistrationMsg
             when ("dataframe") {
                 return attachDataFrameMsg(cmd, msgArgs, st);
             }
+            when ("segarray"){
+                var sa_map: map(string, string);
+                var attParam = new ParameterObj("name", name+"_segments", ObjectType.VALUE, "");
+                var subArgs = new MessageArgs(new list([attParam, ]));
+                sa_map.add("segments", attachMsg(cmd, subArgs, st).msg);
+
+                var attParam2 = new ParameterObj("name", name+"_values", ObjectType.VALUE, "");
+                var subArgs2 = new MessageArgs(new list([attParam2, ]));
+                sa_map.add("values", attachMsg(cmd, subArgs2, st).msg);
+                return new MsgTuple("segarray+%jt".format(sa_map), MsgType.NORMAL); 
+            }
             otherwise {
                 regLogger.warn(getModuleName(),getRoutineName(),getLineNumber(), 
-                            "Unsupported type provided: '%s'. Supported types are: pdarray, strings, categorical, segarray, series, and dataframe".format(dtype));
+                            "Unsupported type provided: '%s'. Supported types are: pdarray, strings, categorical, series, and dataframe".format(dtype));
```

Comment:
segarray should still be one of the supported types right?

Reply:
yeah, updated